paperscraper/utils.py

```diff
@@ -61,26 +60,22 @@ async def _filler(self, rate_limit: float = 1):
             self.rate_limit = rate_limit
             sleep = self._get_sleep()
             updated_at = time.monotonic()
-            fraction = 0
-            extra_increment = 0
-            for i in range(self._queue.maxsize):
-                self._queue.put_nowait(i)
             while True:
-                if not self._queue.full():
-                    now = time.monotonic()
-                    increment = rate_limit * (now - updated_at)
-                    fraction += increment % 1
-                    extra_increment = fraction // 1
-                    items_2_add = int(
-                        min(
-                            self._queue.maxsize - self._queue.qsize(),
-                            int(increment) + extra_increment,
-                        )
-                    )
-                    fraction = fraction % 1
-                    for i in range(items_2_add):
-                        self._queue.put_nowait(i)
-                    updated_at = now
+                now = time.monotonic()
```

Comment:
Imo it's better to use `time.perf_counter` for high-speed stuff, `time.monotonic` is not as high resolution

Reply:
Thanks! Didn't know about that one